From 46895c340c4a188036a1e48b2ba7c48f8311e4b7 Mon Sep 17 00:00:00 2001
From: Vijay kumar Tuamti <vtumati@codeaurora.org>
Date: Mon, 3 Feb 2014 15:36:29 +0530
Subject: [PATCH 102/176] Camera: Increase preview buffer number in ZSL mode by
 2

These two buffers are added for some flexibility
in forming matched super buf in ZSL queue.

Change-Id: Ib91b62b00317ba4e0f9bb85f5be0e72c6767f617
---
 QCamera2/HAL/QCamera2HWI.cpp | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/QCamera2/HAL/QCamera2HWI.cpp b/QCamera2/HAL/QCamera2HWI.cpp
index ec36e55..b30a1b3 100644
--- a/QCamera2/HAL/QCamera2HWI.cpp
+++ b/QCamera2/HAL/QCamera2HWI.cpp
@@ -40,6 +40,7 @@
 
 #define MAP_TO_DRIVER_COORDINATE(val, base, scale, offset) (val * scale / base + offset)
 #define CAMERA_MIN_STREAMING_BUFFERS     3
+#define EXTRA_ZSL_PREVIEW_STREAM_BUF     2
 #define CAMERA_MIN_JPEG_ENCODING_BUFFERS 2
 #define CAMERA_MIN_VIDEO_BUFFERS         9
 
@@ -1412,7 +1413,14 @@ uint8_t QCamera2HardwareInterface::getBufNumRequired(cam_stream_type_t stream_ty
     case CAM_STREAM_TYPE_PREVIEW:
         {
             if (mParameters.isZSLMode()) {
-                bufferCnt = zslQBuffers + minCircularBufNum;
+                /* We need to add two extra streming buffers to add
+                  flexibility in forming matched super buf in ZSL queue.
+                  with number being 'zslQBuffers + minCircularBufNum'
+                  we see preview buffers sometimes get dropped at CPP
+                  and super buf is not forming in ZSL Q for long time. */
+
+                bufferCnt = zslQBuffers + minCircularBufNum +
+                        EXTRA_ZSL_PREVIEW_STREAM_BUF;
             } else {
                 bufferCnt = CAMERA_MIN_STREAMING_BUFFERS +
                             mParameters.getMaxUnmatchedFramesInQueue();
-- 
1.8.3.1

